File: C/Assignment24/Program5.c
//////////////////////////////////////////////////////////////////////////////////////////////////
//  
// File name :   Program5.c
// Description : Write a program which accept string from user & display it in reverse order.
// Author :      Pallavi Omprakash Malewar
// Date :        22/07/2026
//
//////////////////////////////////////////////////////////////////////////////////////////////////

/*

Input :     "MarvellouS"

Output :    "SuollevraM"

*/


// #include<stdio.h>
// #include<stdlib.h>

// void ReverseDisplay(char *str)
// {
//     char *start = NULL;
//     int iCount = 0;

//     start = str;

//     while(*str != '\0')
//     {
//         iCount++;
//         str++;
//     }
//     str--;

//     while(str >= start)
//     {
//         printf("%c", *str);
//         str--;
//     }

//     printf("\n");
// }

// int main()
// {
//     char arr[20];

//     printf("Enter the String : ");
//     scanf("%[^'\n']s", arr);

//     ReverseDisplay(arr);

//     return 0;
// }


#include<stdio.h>
#include<stdlib.h>

void ReverseDisplay(char *str)
{
    char *start = NULL;
    int iCount = 0;

    start = str;

    while(*start != '\0')
    {
        iCount++;
        start++;
    }
    start--;

    while(start >= str)
    {
        printf("%c", *start);
        start--;
    }

    printf("\n");
}

int main()
{
    char arr[20];

    printf("Enter the String : ");
    scanf("%[^'\n']s", arr);

    ReverseDisplay(arr);

    return 0;
}

File: JAVA/Assignment 10/Program5.java
/////////////////////////////////////////////////////////////////////////////////////////
//                                                                                     //
// Problem Statement : Write a Java program which accept expression from user. Check   //
//                     whether the expressions is balanced parenthesis or not.         //
// Author : Pallavi Omprakash Malewar                                                  //
// Date : 4 Aug 2026                                                                   //
//                                                                                     //
/////////////////////////////////////////////////////////////////////////////////////////


/*  

    Input : (a + (f - g) * 2 (a - d))

    Output: TRUE   

    Input : (a + (f - g) * 2 (a - d)

    Output: FALSE   

*/


import java.util.HashMap;
import java.util.Scanner;

class MyString
{
    public boolean CheckParenthesis(String s1)
    {        
        int n = s1.length();
        int iCount = 0;

        for(int i = 0 ; i < n ; i++)
        {
            if(iCount < 0) return false;

            if(s1.charAt(i) == '(') iCount++;
            if(s1.charAt(i) == ')') iCount--;
        }
        if(iCount == 0) return true;
        else return false;
    }
}

class Program5
{
    public static void main(String A[])
    {
        Scanner sobj = new Scanner(System.in);

        System.out.print("Enter String : ");
        String s1 = sobj.nextLine();

        MyString mobj = new MyString();

        boolean bRet = mobj.CheckParenthesis(s1);

        if(bRet == true) System.out.println("TRUE");
        else System.out.println("FALSE");
    }
}
